chapters/stack/functions/reading/README.md

```diff
@@ -0,0 +1,172 @@
+
+```Assembly
+
+foo:
+    push ebp
+    mov ebp, esp
+
+    mov eax, [ebp + 8]
+    mov ebx, [ebp + 12]
+    mov ecx, [ebp + 16]
+
+    add eax, ebx
+    add eax, ecx
+
+    leave
+    ret
+```
+
+### Remarks:
+
+1. A function is defined by a label.
+
+2. After the function's preamble, the stack looks as follows:
+
+   ![stack.svg](../media/stack.svg)
+
+3. Note that during the execution of the function, what does not change is the position of the frame pointer. This is the reason for its name: it points to the current function's frame. Therefore, it is common to access a function's parameters through the frame pointer. Assuming a 32-bit system and processor word-sized parameters (32 bits, 4 bytes), we will have:
+
+   - the first argument is found at address `ebp+8`
+   - the second argument is found at address `ebp+12`
+   - the third argument is found at address `ebp+16`
+   - etc.
+
+   This is why, to get the parameters of the foo function in the eax, ebx, ecx registers, we use the constructions:
+
+```Assembly
+   mov eax, dword [ebp+8]   ; first argument in eax
+   mov ebx, dword [ebp+12]  ; second argument in ebx
+   mov ecx, dword [ebp+16]  ; third argument in ecx
+```
+
+4. The return value of a function is placed in registers (generally in eax).
+   - If the return value is **8 bits**, the function's result is placed in `al`.
+   - If the return value is **16 bits**, the function's result is placed in `ax`.
+   - If the return value is **32 bits**, the function's result is placed in `eax`.
+   - If the return value is **64 bits**, the result is placed in the `edx` and `eax` registers. The most significant 32 bits are placed in `edx`, and the rest in the `eax` register.
+
+    *Additionally, in some cases, a memory address can be returned to the stack/heap (eg. malloc), or other memory areas, which refer to the desired object after the function call.*
+
+5. A function uses the same hardware registers; therefore, when exiting the function, the values of the registers are no longer the same. To avoid this situation, some/all registers can be saved on the stack.
+
+
+> **NOTE:**  Since assembly languages offer more opportunities, there is a need for calling conventions in x86. The difference between them may consist of the parameter order, how the parameters are passed to the function, which registers need to be preserved by the callee or whether the caller or callee handles stack preparation. More details can be found [here](https://en.wikipedia.org/wiki/X86_calling_conventions) or [here](https://levelup.gitconnected.com/x86-calling-conventions-a34812afe097) if Wikipedia is too mainstream for you.
+> For us, the registers `eax`, `ecx`, `edx` are considered **clobbered** (or volatile), and the callee can do whatever it wants to them. On the other hand, the callee has to ensure that `ebx` exits the function with the same value it has entered with.
```

Comment:
I suggest moving this information (what registers we can ovewrite and not) to main-body contents, maybe beneath `5.`. It's important info. Though this is more of an aesthetics change - so I don't feel strongly about it.